### ISSUE TITLE
Add logic for only compressing certain content types

### ISSUE THREAD

**Issue by czechboy0** [czechboy0/gzip-vapor#2]
Interesting article about what does and doesn't make sense to gzip: https://www.fastly.com/blog/new-gzip-settings-and-deciding-what-compress

We need to add logic to look at the response's content type and only compress the ones that deflate well (all forms of text, mostly).

/cc @tannernelson thoughts?

**Comment by tanner0101** [comment]
That's interesting. As long as the middleware is applied late enough in the chain we should have `content-type` information in the headers to help make this decision.

**Comment by czechboy0** [comment]
Hmm, is there a middleware which sees the response but can't see the content type yet? I thought the routes always return a fully defined response. But yeah, we can always choose a sane default (to not gzip unless we know the content type and it's a well compressible one)

**Comment by tanner0101** [comment]
If the response has `var json: JSON?` set, but hasn't gone through the `JSONMiddleware` yet, then it would appear the response is empty. The `JSONMiddleware` is what serializes the `JSON` to the body and sets the appropriate headers, `Content-Length`, `Content-Type` etc. And there are other middlewares that work like this, so gzipping would need to be toward the end of the line.

**Comment by czechboy0** [comment]
Yeah, makes sense. I guess we'd just need to document how to add that
middleware. Works for me great, I'd like to see more people gzipping soon.
